### PR TITLE
#261 [Grid] Add MetaData Filter

### PR DESCRIPTION
**Pull request by martineiber**
## Changes in this pull request
Part of #261

### CODE REVIEW

File: src/DataIndex/Query/AssetQuery.php [hunk 1]
```diff
@@ -97,4 +98,11 @@ public function searchByIds(array $ids): self
 
         return $this;
     }
+
+    public function filterMetaDate(string $name, string $type, mixed $data): self
```
Comment (lukmzig) [inline src/DataIndex/Query/AssetQuery.php]:
```suggestion
    public function filterMetaData(string $name, string $type, mixed $data): self
```

File: src/DataIndex/Filter/Asset/MetaData/TextAreaFilter.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        return $query;
+    }
+
+    private function applyTextAreaFilter(ColumnFilter $column, AssetQuery $query): AssetQuery
+    {
+        if (!is_string($column->getFilterValue())) {
+            throw new InvalidArgumentException('Filter value for textarea must be a string');
+        }
+
+        $query->filterMetaDate($column->getKey(), FilterType::TEXTAREA->value, $column->getFilterValue());
```
Comment (lukmzig) [inline src/DataIndex/Filter/Asset/MetaData/TextAreaFilter.php]:
```suggestion
        $query->filterMetaData($column->getKey(), FilterType::TEXTAREA->value, $column->getFilterValue());
```

File: src/DataIndex/Query/AssetQuery.php (hunk 1 above)
Comment (martineiber) [reply src/DataIndex/Query/AssetQuery.php]: Good Catch 

Fixed here [a376936](https://github.com/pimcore/studio-backend-bundle/pull/277/commits/a3769367344122ff8ec284fbfd6b5a227ee3cb17)

**Review by lukmzig — APPROVED** [review]
image

**Pull request merged**